tests/test_diffraction.py

```diff
@@ -0,0 +1,146 @@
+                    zoom = 1 + 10*np.random.rand()
+                    dp.compute(
+                        system=(box, positions),
+                        view_orientation=view_orientation,
+                        zoom=zoom)
+
+                    # Assert the pixel at the center (k=0) is the maximum value
+                    diff = dp.diffraction
+                    max_index = np.unravel_index(np.argmax(diff), diff.shape)
+                    center_index = (output_size//2, output_size//2)
+                    self.assertEqual(max_index, center_index)
+
+                    # The value at k=0 should be 1 because of normalization
+                    # by (number of points)**2
+                    npt.assert_allclose(dp.diffraction[center_index], 1)
+
+    def test_center_ordered(self):
+        """Assert the center of the image is an intensity peak for an ordered
+        system.
+        """
+        box, positions = freud.data.UnitCell.bcc().generate_system(10)
+
+        # Test different parities (odd/even) of grid_size and output_size
+        for grid_size in [255, 256]:
+            for output_size in [255, 256]:
+                dp = freud.diffraction.DiffractionPattern(
+                    grid_size=grid_size, output_size=output_size)
+                # Use a random view orientation and a random zoom
+                for view_orientation in rowan.random.rand(10):
+                    zoom = 1 + 10*np.random.rand()
+                    dp.compute(
+                        system=(box, positions),
+                        view_orientation=view_orientation,
+                        zoom=zoom)
+
+                    # Assert the pixel at the center (k=0) is the maximum value
+                    diff = dp.diffraction
+                    max_index = np.unravel_index(np.argmax(diff), diff.shape)
+                    center_index = (output_size//2, output_size//2)
+                    self.assertEqual(max_index, center_index)
+
+                    # The value at k=0 should be 1 because of normalization
+                    # by (number of points)**2
+                    npt.assert_allclose(dp.diffraction[center_index], 1)
+
+    def test_repr(self):
+        dp = freud.diffraction.DiffractionPattern()
+        self.assertEqual(str(dp), str(eval(repr(dp))))
+
+        # Use non-default arguments for all parameters
+        dp = freud.diffraction.DiffractionPattern(
+            grid_size=123, output_size=234)
+        self.assertEqual(str(dp), str(eval(repr(dp))))
+
+    def test_k_values_and_k_vectors(self):
+        dp = freud.diffraction.DiffractionPattern()
+        box, positions = freud.data.UnitCell.fcc().generate_system(4)
+        dp.compute((box, positions))
+
+        output_size = dp.output_size
+        npt.assert_allclose(dp.k_values[output_size//2], 0)
+        center_index = (output_size//2, output_size//2)
+        npt.assert_allclose(dp.k_vectors[center_index], [0, 0, 0])
```

Comment:
Perhaps we could run this test using many random systems of different sizes instead of a single fcc system, so the `k_values` and `k_vectors` will have different values across different systems. I think adding this would be a more robust test that the middle `k_vector`/`k_value` is what we want it to be.

Reply:
I am always strongly in favor of adding more tests, particularly of non-toy systems.